docs: Document Pin Messages permission

thehatkid opened the pull request:
Adds documentation for permission flag `PIN_MESSAGES` (1 << 51), allowing pinning and unpinning text channel messages.

thehatkid committed:
Document Pin Messages permission flag

Adds documentation for permission flag `PIN_MESSAGES` (1 << 51),
allowing pinning and unpinning text channel messages.

Lulalaby commented on docs/topics/permissions.md:
```diff
@@ -83,6 +83,7 @@ Below is a table of all current permissions, their integer values in hexadecimal
 | SEND_VOICE_MESSAGES                    | `0x0000400000000000` `(1 << 46)` | Allows sending voice messages                                                                                                                                                                                                                              | T, V, S                    |
 | SEND_POLLS                             | `0x0002000000000000` `(1 << 49)` | Allows sending polls                                                                                                                                                                                                                                       | T, V, S                    |
 | USE_EXTERNAL_APPS                      | `0x0004000000000000` `(1 << 50)` | Allows user-installed apps to send public responses. When disabled, users will still be allowed to use their apps but the responses will be ephemeral. This only applies to apps not also installed to the server.                                         | T, V, S                    |
+| PIN_MESSAGES                           | `0x0008000000000000` `(1 << 51)` | Allows pinning and unpinning messages                                                                                                                                                                                                                      | T                          |
```
```suggestion
| PIN_MESSAGES                           | `0x0008000000000000` `(1 << 51)` | Allows pinning and unpinning messages                                                                                                                                                                                                                      | T, V, S                    |
```

This permission also applies to voice and stage channels cause both have text

advaith1 replied:
those dont seem to support pinning

Lulalaby replied:
Oop. My memory fooled me the. Disregard

anthonydiscord reviewed (changes requested):
Change looks good! As part of this can we update [Pin Message](https://discord.com/developers/docs/resources/message#pin-message) and [Unpin Message](https://discord.com/developers/docs/resources/message#unpin-message) to use the `PIN_MESSAGES ` permission instead of `MANAGE_MESSAGES` now?

thehatkid committed:
Replace with `PIN_MESSAGES` permission

Replaces `MANAGE_MESSAGES` with `PIN_MESSAGES` permission flag for
channel message pin and unpin documentation.
